base/special/expm1.jl

```diff
@@ -0,0 +1,232 @@
+@inline expm1_huge(T::Type{Float64}) = 56.0*expm1_ln2(T)
+expm1_overflow(::Type{Float64}) = 7.09782712893383973096e+02
+@inline expm1_underflow(::Type{Float64}) = 5.551115123125783e-17# 2.0^-54
+# Scaled Q's: Qn_here = 2**n * Qn_above, for R(2*z) where z = hfxs = x*x/2:
+expm1_p(hfxs::Float64) = @horner(hfxs, -3.33333333333331316428e-02, # Q1
+                                        1.58730158725481460165e-03, # Q2
+                                       -7.93650757867487942473e-05, # Q3
+                                        4.00821782732936239552e-06, # Q4
+                                       -2.01099218183624371326e-07) # Q5)
+
+expm1_ln2_hi(::Type{Float32}) = 6.9313812256f-01
+expm1_ln2_lo(::Type{Float32}) = 9.0580006145f-06
+expm1_invln2(::Type{Float32}) = 1.4426950216f0
+expm1_huge(::Type{Float32}) = 18.714973f0 # around 27*ln2, taken from source
+expm1_overflow(::Type{Float32}) = 88.72168f0
+expm1_underflow(::Type{Float32}) = 2.9802322f-8 # 2f0^-25
+
+@inline expm1_ln2(T) = expm1_ln2_hi(T) + expm1_ln2_lo(T)
+
+expm1_big_k(::Type{Float64}) = 1024
+expm1_big_k(::Type{Float32}) = 128
+expm1_small_k(::Type{Float64}) = Int32(20)
+expm1_small_k(::Type{Float32}) = Int32(23)
+
+# Domain [-0.34568, 0.34568], range ~[-6.694e-10, 6.696e-10]:
+# |6 / x * (1 + 2 * (1 / (exp(x) - 1) - 1 / x)) - q(x)| < 2**-30.04
+# Scaled coefficients: Qn_here = 2**n * Qn_for_q (see s_expm1.c):
+@inline expm1_p(hfxs::Float32) = @horner(hfxs, -3.3333212137f-2, # Q1
+                                        1.5807170421f-3) # Q2
+
+@inline expm1_two_pow_mk(::Type{Float32}, k) = reinterpret(Float32, UInt32((0x7f-k)<<23))
+@inline expm1_two_pow_mk(::Type{Float64}, k) = reinterpret(Float64, UInt64(0x3ff00000+((-k)<<20))<<32)
+
+@inline expm1_two_pow_k(::Type{Float32}, k) = reinterpret(Float32, UInt32(0x3f800000+(k<<23)))
+@inline expm1_two_pow_k(::Type{Float64}, k) = reinterpret(Float64, UInt64(0x3ff00000+(k<<20))<<32)
+
+#this is 2^-k not 1-2f0^k
+#@inline expm1_one_m_2_pow_mk(::Type{Float32}, k) = 
+@inline expm1_one_m_2_pow_mk(::Type{Float32}, k) = reinterpret(Float32, UInt32(0x3f800000 - (0x1000000>>k)))
+# 0x00000000 after & comes from lowword(1.0) = unsafe_trunc(UInt32, reinterpret(UInt64, 1.0))
+#@inline expm1_one_m_2_pow_mk(::Type{Float64}, k) = reinterpret(Float64, (UInt64(0x3ff00000 - (0x200000>>k))<<32)|0x00000000)
+@inline expm1_one_m_2_pow_mk(::Type{Float64}, k) = reinterpret(Float64, (UInt64(0x3ff00000 - (0x200000>>k))<<32) | unsafe_trunc(UInt32, reinterpret(UInt64, 1.0)))
+
+@inline expm1_two_pow_big_km1(T::Type{Float64}) = 8.98846567431158e307 # T(2.0)^(expm1_big_k(T) - 1)
+@inline expm1_two_pow_big_km1(T::Type{Float32}) = 1.7014118f38# T(2.0)^(expm1_big_k(T) - 1)
+
+function expm1(x::T) where T<:Union{Float32, Float64}
+    xsign = signbit(x)
+    absx = abs(x)
+ 
+    # filter out huge and non-finite argument
+	if absx >= expm1_huge(T) # quit early for large absolute values of x
+        if isinf(absx)
```

Comment:
The indentation in this file is kind of all over the place. Should be 4-space indents everywhere.

Reply:
I will fix it. Not sure why that happened though...